spec/models/color_scheme_spec.rb

```diff
@@ -190,4 +190,53 @@
       expect(resolved["selected"]).to eq("e9e9e9")
     end
   end
+
+  describe "#diverge_from_remote" do
+    fab!(:theme)
+    fab!(:original_scheme) do
+      Fabricate(
+        :color_scheme,
+        name: "somescheme",
+        theme_id: theme.id,
+        user_selectable: true,
+        color_scheme_colors: [
+          Fabricate(:color_scheme_color, name: "primary", hex: "998877"),
+          Fabricate(:color_scheme_color, name: "secondary", hex: "553322"),
+        ],
+      )
+    end
+
+    it "creates a new scheme with the same colors and sets it as the base scheme" do
+      expect(original_scheme.base_scheme_id).to eq(nil)
+
+      original_scheme.diverge_from_remote
+
+      expect(original_scheme.base_scheme.colors.map { |c| [c.name, c.hex] }.sort_by(&:first)).to eq(
+        original_scheme.colors.map { |c| [c.name, c.hex] }.sort_by(&:first),
+      )
+      expect(original_scheme.base_scheme.theme.id).to eq(theme.id)
```

Comment:
```suggestion
      expect(original_scheme.base_scheme.theme_id).to eq(theme.id)
```

Minor but there is no need to load the theme.